tests/s2n_test.h

```diff
@@ -79,10 +79,14 @@
 
 #define FAIL()      FAIL_MSG("")
 
-#define FAIL_MSG( msg ) do { if (isatty(fileno(stdout))) { \
+#define FAIL_MSG( msg ) do { \
+                          /* isatty will overwrite errno on failure */ \
+                          int real_errno = errno; \
+                          if (isatty(fileno(stdout))) { \
                             fprintf(stdout, "\033[31;1mFAILED test %d\033[0m\n%s (%s line %d)\nError Message: '%s'\n Debug String: '%s'\n System Error: %s (%d)\n", test_count, (msg), __FILE__, __LINE__, s2n_strerror(s2n_errno, "EN"), s2n_debug_str, strerror(errno), errno); \
```

Comment:
Would it be cleaner to always restore `errno` here?

Reply:
maybe, the errno is not rewritten if isatty succeeds, but I can do that for completeness.